homeassistant/components/elv/__init__.py

```diff
@@ -1 +1,41 @@
 """The Elv integration."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.helpers import discovery
+from homeassistant.const import (
+    CONF_NAME, CONF_DEVICE)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_TOTAL_ENERGY_KWH = 'total_energy_kwh'
+
+DOMAIN = 'elv'
+
+NOTIFICATION_ID = 'elv_notification'
+NOTIFICATION_TITLE = 'ELV Setup'
+
+ELV_PLATFORMS = ['switch']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICE): cv.string,
+        vol.Optional(CONF_NAME): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the PCA switch platform."""
+
+    # print(config[DOMAIN].get(CONF_DEVICE))
+    hass.data[DOMAIN] = {
+        'serial_device': config[DOMAIN].get(CONF_DEVICE)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.

Reply:
So this is the last thing I have to fiy now, could you please help me out a litte bit. I want to hand over the serial device definition to the switch platform.
So what is the best way to do it? I wanted to get it from the config inside the switch platform, but there, the config dict is just empty.

Reply:
Either store it in `hass.data` or send it inside `discovery_info`, which currently is just an empty dict, ie the fourth argument below.

Reply:
Thank you for the help! It's really nice to have so much support!

I hope it makes sense now.